File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/HealthIntTest.java
package uk.gov.justice.hmpps.prison.api.resource;

import org.junit.jupiter.api.Test;
import org.springframework.core.ParameterizedTypeReference;
import org.springframework.http.HttpMethod;
import org.springframework.http.RequestEntity;
import uk.gov.justice.hmpps.prison.api.resource.impl.ResourceTest;

import java.util.Map;

import static org.assertj.core.api.Assertions.assertThat;
import static org.springframework.http.HttpStatus.OK;

public class HealthIntTest extends ResourceTest {

    @Test
    public void healthReportsOk() {
        getAndVerifyStatusUp("/health");
    }

    @Test
    public void dbReportsOk() {
        getAndVerifyStatusUp("/health/db");
    }

    @Test
    public void pingReportsOk() {
        getAndVerifyStatusUp("/health/ping");
    }

    @Test
    public void livenessReportsOk() {
        getAndVerifyStatusUp("/health/liveness");
    }

    @Test
    public void readinessReportsOk() {
        getAndVerifyStatusUp("/health/readiness");
    }

    private void getAndVerifyStatusUp(final String url) {
        final var response = testRestTemplate.exchange(
                url,
                HttpMethod.GET,
                RequestEntity.EMPTY,
                new ParameterizedTypeReference<String>() {
                },
                Map.of());

        assertThat(response.getStatusCode()).isEqualTo(OK);
        assertThat(getBodyAsJsonContent(response)).extractingJsonPathStringValue("status").isEqualTo("UP");
    }
}

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/HealthIntTest.kt
package uk.gov.justice.hmpps.prison.api.resource

import org.junit.jupiter.api.Test
import uk.gov.justice.hmpps.prison.api.resource.impl.ResourceTest

class HealthIntTest : ResourceTest() {
  @Test
  fun healthReportsOk() {
    getAndVerifyStatusUp("/health")
  }

  @Test
  fun dbReportsOk() {
    getAndVerifyStatusUp("/health/db")
  }

  @Test
  fun pingReportsOk() {
    getAndVerifyStatusUp("/health/ping")
  }

  @Test
  fun livenessReportsOk() {
    getAndVerifyStatusUp("/health/liveness")
  }

  @Test
  fun readinessReportsOk() {
    getAndVerifyStatusUp("/health/readiness")
  }

  private fun getAndVerifyStatusUp(url: String) {
    webTestClient.get().uri(url).exchange().expectStatus().isOk
      .expectBody().jsonPath("status").isEqualTo("UP")
  }
}
